src/AspNetCore/Src/aspnetcoreconfig.cxx

```diff
@@ -8,16 +8,21 @@ ASPNETCORE_CONFIG::~ASPNETCORE_CONFIG()
     //
     // the destructor will be called once IIS decides to recycle the module context (i.e., application)
     //
+    // shutting down core application first
+    if (ASPNETCORE_APPLICATION::GetInstance() != NULL) {
```

Comment:
This might not be right. Ideally we should be called in response to RecycleApplication being called. So the flow is basically:

RecycleApplication -> Shutdown ASP.NET Core.

Are you sure this is the place that triggers shutdown of dotnet.exe with regular ANCM?